src/touchPopup.js

```diff
@@ -0,0 +1,45 @@
+import '../style/touchpopup.less'
+
+let popup
+
+const createTouchPopup = (container, win=window) => {
+	if (!popup) {
+		popup = win.document.createElement('div')
+		popup.setAttribute('dir', 'ltr')
+		popup.classList.add('wp-touch-popup')
+		popup.style.visibility = 'hidden'
+		container.appendChild(popup)
+  }
+  
+  const popupEvents = {/* onShow, onHide */}
+
+	const show = (content) => {
+		popup.innerHTML = content
+    popup.style.visibility = 'visible'
+    
+    if ( popupEvents.onShow ) {
+			popupEvents.onShow( popup )
+		}
+	}
+
+	const hide = () => {
+    if ( popupEvents.onHide ) {
+			popupEvents.onHide(popup)
+    }
+    
+		popup.style.visibility = 'hidden'
+  }
+  
+  const subscribe = ( events = {} ) => {
+		if ( events.onShow ) {
+			popupEvents.onShow = events.onShow
+		}
+		if ( events.onHide ) {
+			popupEvents.onHide = events.onHide
+		}
+	}
+
+	return { show, hide, subscribe }
+}
+
+export { createTouchPopup }
```

Comment:
Can we have a common popup component, then touchable popup and desktop popup can enhance using the common popup component. so most parts of it are the same.

Reply:
Yea I was thinking along the same lines actually (hence "Opportunities for simplification/refactoring" TODO in PR the description). As a starting point I'm using what Stephane [shared earlier](https://github.com/wikimedia/wikipedia-preview/pull/12#issuecomment-648900604):

>...when we do mobile, we can update "Popup" to handle it or, even better, create another container that acts as a bottom drawer and let the index decide which container to use depending on the platform.

And that line of thinking was already reflected in https://github.com/wikimedia/wikipedia-preview/commit/715045cd2f960cfe5621950c4d5d498c33f67e78 actually. Looking at the codebase now, it seems there's definitely some opportunity for simplification. Already seeing that `show` and `hide` will do different things depending on popup though, so we're probably going to have to make use of `isTouch` in there.

Reply:
What about something like https://github.com/wikimedia/wikipedia-preview/commit/43dc2c1dceb379181f8b28afcc9b07d12ebf4c7a. Sharing in a exploratory sub-branch just in case before committing to official branch, highlights:

* Abstract the common pieces to popup.js, leave the touch/hover specific code to hoverPopup and touchPopup respectively. 
* Simplify styles by getting rid of touchpopup.less altogether and including variable for `popup` z-index along with all popups styles directly in popup.less 
* Have `isTouch` return a variable instead of function

Reply:
> * Simplify styles by getting rid of touchpopup.less altogether and including variable for `popup` z-index along with all popups styles directly in popup.less
> * Have `isTouch` return a variable instead of function

Let's take these two into this branch.

> * Abstract the common pieces to popup.js, leave the touch/hover specific code to hoverPopup and touchPopup respectively.

Although I said this simplication/refactoring in the beginning, can we wait and see until more components (ex. gallery), so we have the full picture to see how we do the refactoring here.

Reply:
Yea that sounds good. It's why I shared in a different branch, the refactoring would be somewhat significant